targetFunctions/common/groupLayers/groupLayersMod.m

```diff
@@ -1,72 +1,47 @@
-% Arrange layers according to geometry and apply any coverage correction. The paratt calculation proceeds through the 
-% z,rho,rough stack, and the parameter 'ssub' is the final roughness encountered. 
+function [outputLayers, ssubs] = groupLayersMod(layers,subRoughs,geometry)
+% Arrange layers according to geometry.
+% The paratt calculation proceeds through the z,rho,rough stack; and the
+% parameter 'ssub' is the final roughness encountered.
 %
-% * For air liquid 'ssub' is therefore the substrate roughness.
-% * For solid liquid, the substrate roughness is the first roughness encountered, and 'ssub' is then the roughness of the outermost layer
+% * For air/liquid 'ssub' is therefore the substrate roughness.
+% * For solid/liquid, the substrate roughness is the first roughness
+%   encountered, and 'ssub' is then the roughness of the outermost layer.
 %
 % USAGE::
 %
-%     [outLayers, ssubs] = groupLayersMod(resampledLayers,subRoughs,geometry,bulkIns,bulkOuts)
+%     [outLayers, ssubs] = groupLayersMod(resampledLayers,subRoughs,geometry)
 %
 % INPUTS:
-%     * resampledLayers: cell array, one for each contrast. Each cell is the list of layer values for each contrast.
-%     * subRoughs:  Double of substrate roughness for each contrast.
+%     * layers: List of layer values for this contrast.
+%     * subRoughs:  Double of substrate roughness for this contrast.
 %     * geometry: 'Air / Liquid (or solid)' or 'Solid / Liquid'
-%     * bulkIns: vector of bulkIn values.
-%     * bulkOuts: vector of bulkOut values.
 %
 % Outputs:
-%     * outLayers: cell array of layers param values for each contrast.
+%     * outputLayers: Layers parameter values for this contrast.
 %     * ssubs: vector of ssub values.
 
 ssubs = subRoughs;
-layers = zeros(size(resampledLayers));
-if ~isempty(resampledLayers)
+
+if ~isempty(layers)
+    
     if strcmpi(geometry, coderEnums.geometryOptions.AirSubstrate)
-            layers = resampledLayers;
+            outputLayers = layers;
     else
-            roughs = resampledLayers(:,4);
-            sldss = resampledLayers(:,2:3);
-            thicks = resampledLayers(:,1);
-            rsub = roughs(end);
-            if length(roughs) > 1
-                roughs = [ssubs ; roughs(1:end-1)];
+            roughnesses = layers(:,4);
+            rsub = roughnesses(end);
+            if length(roughnesses) > 1
+                roughnesses = [ssubs ; roughnesses(1:end-1)];
             else
-                roughs = ssubs;
-            end
-            n = size(resampledLayers,2);
-            if n == 5
-                cov = resampledLayers(:,5);
-                layers = [thicks sldss roughs cov];
-            else
-                layers = [thicks sldss roughs];
+                roughnesses = ssubs;
             end
+            outputLayers = [layers(:, 1:3) roughnesses layers(:, 5:end)];
             ssubs = rsub;
     end
-    
-    % Deal with the %coverage if present
-    n = size(resampledLayers,2);
```

Comment:
This looks similar to `applyHydration` is that why its gone or did you relocate?

Reply:
Yes, this is a duplicate of `applyHydration`. I therefore removed this code and placed a call to `applyHydration` in `coreLayersCalculation`. This then meant that `applyHydration` was being called twice for all custom layers calculations, allowing it to be removed from `processCustomFunction`. This then enabled the `processCustomFunction` routines to be merged.